pandas/core/arrays/string_.py

```diff
@@ -282,8 +284,25 @@ def astype(self, dtype, copy=True):
         return super().astype(dtype, copy)
 
     def _reduce(self, name, skipna=True, **kwargs):
+        if name in ["min", "max"]:
+            return getattr(self, name)(skipna=skipna, **kwargs)
+
         raise TypeError(f"Cannot perform reduction '{name}' with string dtype")
 
+    def min(self, skipna=True, **kwargs):
+        nv.validate_min((), kwargs)
```

Comment:
Can you add a `if kwargs` so we only call this if there are actually kwargs (which avoids the overhead of this there are actually no kwargs

Reply:
we don’t do this anywhere else so -1 on skipping the check here (this overhead is tiny anyhow); pls don’t prematurely optimize things 

it makes the code much harder to follow and read